models/eos/bgp_global/eos_bgp_global.yaml

```diff
@@ -0,0 +1,1327 @@
+                    preserve:
+                      description: preserve route attributes, overwriting route-map changes
+                      type: bool 
+                transport:
+                  description: Configure transport port for TCP session
+                  type: int
+            default_metric:
+              description: Default metric.
+              type: int
+            distance:
+              description: Define an administrative distance.
+              type: dict
+              suboptions:
+                external:
+                  descriptions: distance for external routes.
+                  type: int
+                internal:
+                  description: distance for internal routes.
+                  type: int
+                local:
+                  description: distance for local routes.
+                  type: int
+            graceful_restart:
+              description: Enable graceful restart mode.
+              type: dict
+              suboptions:
+                set:
+                  description: When True, graceful restart is set.
+                  type: bool
+                restart_time:
+                  description: Set the max time needed to restart and come back up.
+                  type: int
+                stalepath_time:
+                  decription: Set the max time to hold onto restarting peer stale paths.
+                  type: int
+            graceful_restart_helper:
+              description: Enable graceful restart helper mode.
+              type: bool
+            access_group:
+              desscription: ip/ipv6 access list configuration.
+              type: dict
+              suboptions:
+                afi:
+                  desscription: Specify ip/ipv6.
+                  type: str
+                  choices: ['ip', 'ipv6']
+                acl_name:
+                  description: access list name.
+                  type: str
+                direction:
+                  description: direction of packets.
+                  type: str
+            maximum_paths:
+              description: Maximum number of equal cost paths.
+              type: dict
+              suboptions:
+                max_equal_cost_paths:
+                  description: Value for maximum number of equal cost paths.
+                  type: int
+                max_installed_ecmp_paths:
+                  description: Value for maximum number of installed ECMP routes.
+                  type: int
+            neighbor:
+              description: Configure routing for a network.
+              type: list
+              elements: dict
+              suboptions:
+                peer:
+                  type: str
+                  description: Neighbor address or peer group.
+                additional_paths:
+                  description: BGP additional-paths commands.
+                  type: str
+                  choices: ['send', 'receive']
+                allowas_in:
```

Comment:
we can avoid duplication of "allowas_in" dict by defining it once and reuse it whenever required.
if possible